trade-finance-manager-ui/server/api.test.js

```diff
@@ -15,3 +17,131 @@ describe('getFacilities()', () => {
+
+    const queryParams = { page: 0 };
+    const token = 'testToken';
+    const response = await api.getDeals(queryParams, token);
+
+    expect(axios).toHaveBeenCalledTimes(1);
+    expect(axios).toHaveBeenCalledWith({
+      method: 'get',
+      url: `${TFM_API_URL}/v1/deals`,
+      headers: {
+        Authorization: token,
+        'Content-Type': 'application/json',
+        'x-api-key': TFM_API_KEY,
+      },
+      params: queryParams,
+    });
+    expect(response).toEqual({
+      deals: [{ id: 1, name: 'Deal 1' }, { id: 2, name: 'Deal 2' }],
+      pagination: { totalItems: 2, currentPage: 0, totalPages: 1 },
+    });
+  });
+
+  it('should return an empty deals array and pagination metadata with total items and pages equal to 0 when TFM API does not return data', async () => {
+    const mockResponse = {};
+    axios.mockResolvedValueOnce(mockResponse);
+
+    const queryParams = { page: 1 };
+    const token = 'testToken';
+    const response = await api.getDeals(queryParams, token);
+
+    expect(axios).toHaveBeenCalledTimes(1);
+    expect(axios).toHaveBeenCalledWith({
+      method: 'get',
+      url: `${TFM_API_URL}/v1/deals`,
+      headers: {
+        Authorization: token,
+        'Content-Type': 'application/json',
+        'x-api-key': TFM_API_KEY,
+      },
+      params: queryParams,
+    });
+    expect(response).toEqual({
+      deals: [],
+      pagination: { totalItems: 0, currentPage: 1, totalPages: 0 },
+    });
+  });
+
+  it('should return an empty object when getting the deals from TFM API throws an error', async () => {
+    const error = new Error('API Error');
+    const consoleErrorMock = jest.spyOn(console, 'error').mockImplementation();
+    axios.mockRejectedValueOnce(error);
+
+    const queryParams = { page: 0 };
+    const token = 'testToken';
+    const response = await api.getDeals(queryParams, token);
+
+    expect(axios).toHaveBeenCalledTimes(1);
+    expect(axios).toHaveBeenCalledWith({
+      method: 'get',
+      url: `${TFM_API_URL}/v1/deals`,
+      headers: {
+        Authorization: token,
+        'Content-Type': 'application/json',
+        'x-api-key': TFM_API_KEY,
+      },
+      params: queryParams,
+    });
+    expect(consoleErrorMock).toHaveBeenCalledWith('Unable to get deals %O', error);
+    expect(response).toEqual({});
+  });
+
+  it('should return an empty object when TFM API returns an unexpected response format', async () => {
+    const mockResponse = { data: 'invalidData' };
+    axios.mockResolvedValueOnce(mockResponse);
+
+    const queryParams = { page: 0 };
+    const token = 'testToken';
+    const response = await api.getDeals(queryParams, token);
+
+    expect(axios).toHaveBeenCalledTimes(1);
+    expect(axios).toHaveBeenCalledWith({
+      method: 'get',
+      url: `${TFM_API_URL}/v1/deals`,
+      headers: {
+        Authorization: token,
+        'Content-Type': 'application/json',
+        'x-api-key': TFM_API_KEY,
+      },
+      params: queryParams,
```

Comment:
Could we create an e.g `baseExpected` for everything but `params`? Seems to be duplicated throughout

Reply:
Great spot, I've done this :)